config.go

```diff
@@ -0,0 +1,174 @@
+type Config struct {
+	// Config from the service config file
+	Collector         CollectorConfig `yaml:"collector"`
+	Producers         ProducersConfig `yaml:"producers"`
+	IAMConfigPath     string          `yaml:"iam_config_path"`
+	CACertificatePath string          `yaml:"ca_certificate_path"`
+	VersionFlag       bool
+
+	// Generated by dcos.NodeInfo{}
+	MesosID   string
+	IPAddress string
+	ClusterID string
+
+	// Flag configuration
+	DCOSRole   string
+	ConfigPath string
+	LogLevel   string
+}
+
+// CollectorConfig contains configuration options relevant to the "collector"
+// portion of this project. That is, the code responsible for querying Mesos,
+// et. al to gather metrics and send them to a "producer".
+type CollectorConfig struct {
+	HTTPProfiler  bool   `yaml:"http_profiler"`
+	IPCommand     string `yaml:"ip_command"`
+	PollingPeriod int    `yaml:"polling_period"`
+
+	MasterConfig collector.MasterConfig `yaml:"master_config,omitempty"`
+	AgentConfig  collector.AgentConfig  `yaml:"agent_config,omitempty"`
+}
+
+// ProducersConfig contains references to other structs that provide individual producer configs.
+// The configuration for all producers is then located in their corresponding packages.
+//
+// For example: Config.Producers.KafkaProducerConfig references kafkaProducer.Config. This struct
+// contains an optional Kafka configuration. This configuration is available in the source file
+// 'producers/kafka/kafka.go'. It is then the responsibility of the individual producers to
+// validate the configuration the user has provided and panic if necessary.
+type ProducersConfig struct {
+	HTTPProducerConfig httpProducer.Config `yaml:"http,omitempty"`
+	//KafkaProducerConfig  kafkaProducer.Config  `yaml:"kafka,omitempty"`
+	//StatsdProducerConfig statsdProducer.Config `yaml:"statsd,omitempty"`
+}
+
+func (c *Config) setFlags(fs *flag.FlagSet) {
+	fs.StringVar(&c.ConfigPath, "config", c.ConfigPath, "The path to the config file.")
+	fs.StringVar(&c.LogLevel, "loglevel", c.LogLevel, "Logging level (default: info). Must be one of: debug, info, warn, error, fatal, panic.")
+	fs.StringVar(&c.DCOSRole, "role", c.DCOSRole, "The DC/OS role this instance runs on.")
+	fs.BoolVar(&c.VersionFlag, "version", c.VersionFlag, "Print version and revsion then exit")
+}
+
+func (c *Config) loadConfig() error {
+	fileByte, err := ioutil.ReadFile(c.ConfigPath)
+	if err != nil {
+		return err
+	}
+
+	if err = yaml.Unmarshal(fileByte, &c); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Config) getNodeInfo() error {
+	client, err := getClient(c.CACertificatePath, c.IAMConfigPath)
+	if err != nil {
+		return err
+	}
+
+	// Get NodeInfo
+	nodeInfo, err := nodeutil.NewNodeInfo(client)
+	if err != nil {
+		return err
+	}
+
+	ip, err := nodeInfo.DetectIP()
+	if err != nil {
+		return err
+	}
+	c.IPAddress = ip.String()
+
+	id, err := nodeInfo.MesosID(nil)
+	if err != nil {
+		c.MesosID = ""
+	} else {
+		c.MesosID = id
+	}
```

Comment:
if you don't handle error maybe this would be better to read
```go
c.MesosID, err = nodeInfo.MesosID(nil)
if err != nil {
  logrus.Errorf("Could not get mesos ID: %s", err)
}
```